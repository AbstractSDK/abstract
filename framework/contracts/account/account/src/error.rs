use cosmwasm_std::StdError;

use manager::error::ManagerError;
use proxy::error::ProxyError;
use thiserror::Error;

#[derive(Error, Debug, PartialEq)]
pub enum AccountError {
    #[error("{0}")]
    Std(#[from] StdError),

    // #[error("{0}")]
    // Abstract(#[from] AbstractError),

    // #[error("{0}")]
    // AbstractSdk(#[from] AbstractSdkError),

    // #[error("{0}")]
    // Admin(#[from] AdminError),
    #[error(transparent)]
    Manager(#[from] ManagerError),

    #[error(transparent)]
    Proxy(#[from] ProxyError),
}
