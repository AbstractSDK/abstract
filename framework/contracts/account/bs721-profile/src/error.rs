use cosmwasm_std::StdError;
use cw_controllers::AdminError;
use cw_ownable::OwnershipError;
use cw_utils::PaymentError;
use thiserror::Error;

#[derive(Error, Debug)]
pub enum ContractError {
    #[error("{0}")]
    Std(#[from] StdError),

    #[error("{0}")]
    OwnershipError(#[from] OwnershipError),
    #[error("{0}")]
    Payment(#[from] PaymentError),

    #[error("{0}")]
    Admin(#[from] AdminError),

    #[error("{0}")]
    Base(#[from] bs721_base::ContractError),

    #[error("NameNotFound")]
    NameNotFound {},

    #[error("AddressAlreadyMapped")]
    AddressAlreadyMapped {},

    #[error("RecordNameAlreadyExists")]
    RecordNameAlreadyExists {},

    #[error("RecordNameEmpty")]
    RecordNameEmpty {},

    #[error("RecordNameTooLong")]
    RecordNameTooLong {},

    #[error("RecordValueTooLong")]
    RecordValueTooLong {},

    #[error("RecordValueEmpty")]
    RecordValueEmpty {},

    #[error("UnauthorizedVerification")]
    UnauthorizedVerification {},

    #[error("Invalid Metadata")]
    InvalidMetadata {},

    #[error("Unauthorized: Not contract creator or admin")]
    UnauthorizedCreatorOrAdmin {},

    #[error("TooManyRecords max: {max}")]
    TooManyRecords { max: u32 },

    #[error("NotImplemented")]
    NotImplemented {},
}
