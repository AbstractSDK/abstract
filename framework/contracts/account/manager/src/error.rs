use abstract_core::{
    objects::{validation::ValidationError, version_control::VersionControlError},
    AbstractError,
};
use abstract_sdk::{core::objects::module::ModuleInfo, AbstractSdkError};
use cosmwasm_std::{Instantiate2AddressError, StdError};
use cw_controllers::AdminError;
use thiserror::Error;

#[derive(Error, Debug, PartialEq)]
pub enum ManagerError {
    #[error("{0}")]
    Std(#[from] StdError),

    #[error("{0}")]
    Abstract(#[from] AbstractError),

    #[error("{0}")]
    AbstractSdk(#[from] AbstractSdkError),

    #[error("{0}")]
    Admin(#[from] AdminError),

    #[error("{0}")]
    Validation(#[from] ValidationError),

    #[error("{0}")]
    Ownership(#[from] cw_ownable::OwnershipError),

    #[error("{0}")]
    Instantiate2AddressError(#[from] Instantiate2AddressError),

    #[error("{0}")]
    VersionControlError(#[from] VersionControlError),

    #[error("Module with id: {0} is already installed")]
    ModuleAlreadyInstalled(String),

    #[error("Cannot remove module because {0:?} depend(s) on it.")]
    ModuleHasDependents(Vec<String>),

    #[error("Contract got an unexpected Reply")]
    UnexpectedReply(),

    #[error("The name of the proposed module can not have length 0.")]
    InvalidModuleName {},

    #[error("Registering module fails because caller is not module factory")]
    CallerNotModuleFactory {},

    #[error("A migrate msg is required when when migrating this module")]
    MsgRequired {},

    #[error("{0} not upgradable")]
    NotUpgradeable(ModuleInfo),

    #[error("Cannot migrate {} twice", module_id)]
    DuplicateModuleMigration { module_id: String },

    #[error("Your account is currently suspended")]
    AccountSuspended {},

    #[error("The provided contract version {0} is lower than the current version {1}")]
    OlderVersion(String, String),

    #[error("The provided module {0} was not found")]
    ModuleNotFound(String),

    #[error("Module {module_id} with version {version} does not fit requirement {comp}, post_migration: {post_migration}")]
    VersionRequirementNotMet {
        module_id: String,
        version: String,
        comp: String,
        post_migration: bool,
    },

    #[error("module {0} is a dependency of {1} and is not installed.")]
    DependencyNotMet(String, String),

    #[error("The provided module {0} has an invalid module reference.")]
    InvalidReference(ModuleInfo),

    #[error("Cannot remove proxy")]
    CannotRemoveProxy {},

    #[error("No updates were included")]
    NoUpdates {},

    #[error("invalid configuration action, {}", error)]
    InvalidConfigAction { error: StdError },

    #[error("Must use ProposeOwner to change owner")]
    MustUseProposeOwner {},

    #[error("The address {0} doesn't have an owner, the manager can't determine admin right")]
    NoContractOwner(String),

    #[error("
            Checking the admin recursively failed. 
            You either have the an error in your sub-account configuration or you are not authorized to make this call.
    ")]
    SubAccountAdminVerification,

    #[error("Removing sub account failed")]
    SubAccountRemovalFailed {},

    #[error("Register of sub account failed")]
    SubAccountRegisterFailed {},

    #[error("Can't renounce account, with active sub account")]
    RenounceWithSubAccount {},

    #[error("Can't propose Renounced governance, use update_ownership instead")]
    ProposeRenounced {},

    #[error("Can't create account with Renounced governance")]
    InitRenounced {},
}
