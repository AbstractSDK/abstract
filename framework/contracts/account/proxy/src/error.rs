use abstract_core::{objects::ans_host::AnsHostError, AbstractError};
use abstract_sdk::AbstractSdkError;
use cosmwasm_std::{StdError, Uint128};
use cw_asset::AssetError;
use cw_utils::ParseReplyError;
use thiserror::Error;

#[derive(Error, Debug, PartialEq)]
pub enum ProxyError {
    #[error("{0}")]
    Std(#[from] StdError),

    #[error("{0}")]
    Abstract(#[from] AbstractError),

    #[error("{0}")]
    AbstractSdk(#[from] AbstractSdkError),

    #[error("{0}")]
    Asset(#[from] AssetError),

    #[error(transparent)]
    Admin(#[from] ::cw_controllers::AdminError),

    #[error("{0}")]
    Parse(#[from] ParseReplyError),

    #[error("{0}")]
    AnsHostError(#[from] AnsHostError),

    #[error("Module with address {0} is already whitelisted")]
    AlreadyWhitelisted(String),

    #[error("Module with address {0} not found in whitelist")]
    NotWhitelisted(String),

    #[error("Sender is not whitelisted")]
    SenderNotWhitelisted {},

    #[error("Max amount of assets registered")]
    AssetsLimitReached,

    #[error("Max amount of modules registered")]
    ModuleLimitReached,

    #[error("no base asset registered on proxy")]
    MissingBaseAsset,

    #[error("The proposed update resulted in a bad configuration: {0}")]
    BadUpdate(String),

    #[error(
        "Account balance too low, {} requested but it only has {}",
        requested,
        balance
    )]
    Broke {
        balance: Uint128,
        requested: Uint128,
    },

    #[error("Contract got an unexpected Reply")]
    UnexpectedReply(),
}
