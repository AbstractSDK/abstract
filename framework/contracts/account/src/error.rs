use abstract_sdk::std::objects::module::ModuleInfo;
use abstract_std::{
    objects::{registry::RegistryError, validation::ValidationError},
    AbstractError,
};
use cosmwasm_std::{Instantiate2AddressError, StdError};
use thiserror::Error;

#[derive(Error, Debug, PartialEq)]
pub enum AccountError {
    #[error(transparent)]
    Std(#[from] StdError),

    #[error(transparent)]
    Abstract(#[from] AbstractError),

    #[error(transparent)]
    Validation(#[from] ValidationError),

    #[error(transparent)]
    Ownership(#[from] abstract_std::objects::ownership::GovOwnershipError),

    #[error(transparent)]
    Instantiate2AddressError(#[from] Instantiate2AddressError),

    #[error(transparent)]
    RegistryError(#[from] RegistryError),

    #[error("Your account is currently suspended")]
    AccountSuspended {},

    // ** Modules Error ** //
    #[error("Failed to query modules to install: {error}")]
    QueryModulesFailed { error: RegistryError },

    #[error("Module with id: {0} is already installed")]
    ModuleAlreadyInstalled(String),

    #[error("Reinstalls of same version of app or standalone are not allowed")]
    ProhibitedReinstall {},

    #[error("The provided module {0} can't be installed on an Abstract account")]
    ModuleNotInstallable(String),

    #[error("The name of the proposed module can not have length 0.")]
    InvalidModuleName {},

    #[error("The provided module {0} was not found")]
    ModuleNotFound(String),

    #[error("Cannot migrate {} twice", module_id)]
    DuplicateModuleMigration { module_id: String },

    #[error("{0} not upgradable")]
    NotUpgradeable(ModuleInfo),

    #[error("Cannot remove module because {0:?} depend(s) on it.")]
    ModuleHasDependents(Vec<String>),

    #[error("Module {module_id} with version {version} does not fit requirement {comp}, post_migration: {post_migration}")]
    VersionRequirementNotMet {
        module_id: String,
        version: String,
        comp: String,
        post_migration: bool,
    },

    #[error("module {0} is a dependency of {1} and is not installed.")]
    DependencyNotMet(String, String),

    #[error("Max amount of modules registered")]
    ModuleLimitReached {},

    #[error("Module with address {0} is already whitelisted")]
    AlreadyWhitelisted(String),

    #[error("can't remove module that is not whitelisted")]
    NotWhitelisted {},

    // ** Sub Account ** //
    #[error("Removing sub account failed")]
    SubAccountRemovalFailed {},

    #[error("Register of sub account failed")]
    SubAccountRegisterFailed {},

    #[error("Can't renounce account, with active sub account")]
    RenounceWithSubAccount {},

    // ** Other Errors ** //
    #[error("No updates were included")]
    NoUpdates {},

    #[error("invalid configuration action, {}", error)]
    InvalidConfigAction { error: StdError },

    #[error("The provided contract version {0} is lower than the current version {1}")]
    OlderVersion(String, String),

    #[error("Sender is not whitelisted and is not a valid owner")]
    SenderNotWhitelistedOrOwner {},

    #[error("Contract got an unexpected Reply")]
    UnexpectedReply(),

    #[error("The caller ({caller}) is not the owner account's account ({account}). Only account can create sub-accounts for itself.", )]
    SubAccountCreatorNotAccount { caller: String, account: String },

    #[error("You can't chain admin calls")]
    CantChainAdminCalls {},

    #[error("Abstract Account Address don't match to the Contract address")]
    AbsAccInvalidAddr {
        abstract_account: String,
        contract: String,
    },

    #[error("Abstract Account don't have Authentication")]
    AbsAccNoAuth {},

    #[cfg(feature = "xion")]
    #[error(transparent)]
    AbstractXion(#[from] abstract_xion::error::ContractError),
}
