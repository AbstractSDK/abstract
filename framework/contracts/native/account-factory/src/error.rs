use abstract_core::{objects::version_control::VersionControlError, AbstractError};
use abstract_sdk::AbstractSdkError;
use cosmwasm_std::{Instantiate2AddressError, StdError};
use cw_asset::AssetError;
use cw_controllers::AdminError;
use thiserror::Error;

#[derive(Error, Debug, PartialEq)]
pub enum AccountFactoryError {
    #[error("{0}")]
    Std(#[from] StdError),

    #[error("{0}")]
    Abstract(#[from] AbstractError),

    #[error("{0}")]
    AbstractSdk(#[from] AbstractSdkError),

    #[error("{0}")]
    Asset(#[from] AssetError),

    #[error("{0}")]
    Ownership(#[from] cw_ownable::OwnershipError),

    #[error("{0}")]
    Admin(#[from] AdminError),

    #[error("{0}")]
    Instantiate2AddressError(#[from] Instantiate2AddressError),

    #[error("{0}")]
    VersionControlError(#[from] VersionControlError),

    #[error("Contract got an unexpected Reply")]
    UnexpectedReply(),

    #[error("module {0} is required to be of kind {1}")]
    WrongModuleKind(String, String),

    #[error("Your payment does not match the required payment {0}")]
    WrongAmount(String),

    #[error("No payment received")]
    NoPaymentReceived {},

    #[error("Can not create remote accounts without configured IBC host.")]
    IbcHostNotSet {},

    #[error("A trace must exist of at least one or at most {0} hops but has {1}")]
    InvalidTrace(usize, usize),

    #[error("Sender {0} is not the IBC host {1}")]
    SenderNotIbcHost(String, String),
}
