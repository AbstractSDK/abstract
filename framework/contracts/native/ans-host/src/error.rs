use abstract_sdk::AbstractSdkError;
use abstract_std::AbstractError;
use cosmwasm_std::StdError;
use cw_asset::AssetError;
use thiserror::Error;

#[derive(Error, Debug, PartialEq)]
pub enum AnsHostError {
    #[error("{0}")]
    Std(#[from] StdError),

    #[error("{0}")]
    Abstract(#[from] AbstractError),

    #[error("{0}")]
    AbstractSdk(#[from] AbstractSdkError),

    #[error("{0}")]
    Asset(#[from] AssetError),

    #[error("{0}")]
    Ownership(#[from] cw_ownable::OwnershipError),

    #[error("You must provide exactly two assets when adding liquidity")]
    NotTwoAssets {},

    #[error("{} is not part of the provided pool", id)]
    NotInPool { id: String },

    #[error("{} assets is not within range [{}-{}]", provided, min, max)]
    InvalidAssetCount {
        min: usize,
        max: usize,
        provided: usize,
    },

    #[error("Dex {} is not registered", dex)]
    UnregisteredDex { dex: String },

    #[error("Asset {} is not registered", asset)]
    UnregisteredAsset { asset: String },

    #[error("Dex {} is already registered", dex)]
    DexAlreadyRegistered { dex: String },
}
