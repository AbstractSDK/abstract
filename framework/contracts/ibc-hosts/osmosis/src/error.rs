use abstract_core::AbstractError;
use abstract_ibc_host::HostError;
use abstract_sdk::AbstractSdkError;
use cosmwasm_std::StdError;
use dex::error::DexError;
use thiserror::Error;

#[derive(Error, Debug, PartialEq)]
pub enum OsmoError {
    #[error("{0}")]
    Std(#[from] StdError),

    #[error("{0}")]
    DexError(#[from] DexError),

    #[error("{0}")]
    Abstract(#[from] AbstractError),

    #[error("{0}")]
    AbstractSdk(#[from] AbstractSdkError),

    #[error("{0}")]
    HostError(#[from] HostError),

    #[error("DEX {0} is not a known dex on this network.")]
    UnknownDex(String),

    #[error("Cw1155 is unsupported.")]
    Cw1155Unsupported,

    #[error("Can't provide liquidity less than two assets")]
    TooFewAssets {},

    #[error("Can't provide liquidity with more than {0} assets")]
    TooManyAssets(u8),

    #[error("Provided asset {0} not in pool with assets {1:?}.")]
    ArgumentMismatch(String, Vec<String>),

    #[error("Pair {0} on DEX {1} does not match with pair address {2}")]
    DexMismatch(String, String, String),
}
