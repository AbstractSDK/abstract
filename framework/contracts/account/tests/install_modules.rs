use abstract_account::error::AccountError;
use abstract_integration_tests::{create_default_account, mock_modules, AResult};
use abstract_interface::{Abstract, VCExecFns};
use abstract_std::{
    account::{
        ExecuteMsg as AccountMsg, ModuleAddressesResponse, ModuleInstallConfig,
        QueryMsg as AccountQuery,
    },
    objects::module::ModuleInfo,
};
use abstract_testing::prelude::{mock_bech32_admin, TEST_ACCOUNT_ID, TEST_NAMESPACE};
use cw_orch::{prelude::*, take_storage_snapshot};
use mock_modules::{adapter_1, deploy_modules, V1};

#[test]
fn cannot_reinstall_module() -> AResult {
    let chain = MockBech32::new("mock");
    let sender = chain.sender_addr();
    let abstr = Abstract::deploy_on(chain.clone(), mock_bech32_admin(&chain))?;
    let account = create_default_account(&sender, &abstr)?;

    abstr
        .version_control
        .claim_namespace(TEST_ACCOUNT_ID, TEST_NAMESPACE.to_string())?;

    deploy_modules(&chain);

    account.execute(
        &AccountMsg::InstallModules {
            modules: vec![ModuleInstallConfig::new(
                ModuleInfo::from_id(adapter_1::MOCK_ADAPTER_ID, V1.into()).unwrap(),
                None,
            )],
        },
        &[],
    )?;

    let err = account
        .execute(
            &AccountMsg::InstallModules {
                modules: vec![ModuleInstallConfig::new(
                    ModuleInfo::from_id(adapter_1::MOCK_ADAPTER_ID, V1.into()).unwrap(),
                    None,
                )],
            },
            &[],
        )
        .unwrap_err();
    let account_err: AccountError = err.downcast().unwrap();
    assert_eq!(
        account_err,
        AccountError::ModuleAlreadyInstalled(adapter_1::MOCK_ADAPTER_ID.to_owned())
    );
    Ok(())
}

#[test]
fn adds_module_to_account_modules() -> AResult {
    let chain = MockBech32::new("mock");
    let sender = chain.sender_addr();
    let abstr = Abstract::deploy_on(chain.clone(), mock_bech32_admin(&chain))?;
    let account = create_default_account(&sender, &abstr)?;

    abstr
        .version_control
        .claim_namespace(TEST_ACCOUNT_ID, TEST_NAMESPACE.to_string())?;

    deploy_modules(&chain);

    account.execute(
        &AccountMsg::InstallModules {
            modules: vec![ModuleInstallConfig::new(
                ModuleInfo::from_id(adapter_1::MOCK_ADAPTER_ID, V1.into()).unwrap(),
                None,
            )],
        },
        &[],
    )?;

    let addrs: ModuleAddressesResponse = account.query(&AccountQuery::ModuleAddresses {
        ids: vec![adapter_1::MOCK_ADAPTER_ID.to_owned()],
    })?;
    assert_eq!(addrs.modules.len(), 1);
    take_storage_snapshot!(chain, "adds_module_to_account_modules");
    Ok(())
}

#[test]
fn useful_error_module_not_found() -> AResult {
    let chain = MockBech32::new("mock");
    let sender = chain.sender_addr();
    let abstr = Abstract::deploy_on(chain.clone(), mock_bech32_admin(&chain))?;
    let account = create_default_account(&sender, &abstr)?;

    let err = account
        .execute(
            &AccountMsg::InstallModules {
                modules: vec![ModuleInstallConfig::new(
                    ModuleInfo::from_id(adapter_1::MOCK_ADAPTER_ID, V1.into()).unwrap(),
                    None,
                )],
            },
            &[],
        )
        .unwrap_err();

    let account_error: AccountError = err.downcast().unwrap();
    assert!(matches!(
        account_error,
        AccountError::QueryModulesFailed { .. }
    ));
    Ok(())
}
