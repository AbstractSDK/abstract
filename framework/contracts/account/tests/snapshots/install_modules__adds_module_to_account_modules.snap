---
source: contracts/account/tests/install_modules.rs
expression: all_storage
---
"abstract:account-local-0":
  - - aa
    - "false"
  - - ab
    - "{\"name\":\"Default Abstract Account\"}"
  - - af
    - "[]"
  - - ag
    - "{\"trace\":\"local\",\"seq\":0}"
  - - ai
    - "[]"
  - - contract_info
    - "{\"contract\":\"abstract:account\",\"version\":\"0.24.1\"}"
  - - ownership
    - "{\"owner\":{\"monarchy\":{\"monarch\":\"mock14cl2dthqamgucg9sfvv4relp3aa83e40yfzzc8\"}},\"pending_owner\":null,\"pending_expiry\":null}"
"abstract:account-local-1":
  - - "\u0000\u0002actester:mock-adapter1"
    - "\"mock1ds5m6wwuu0cr35kmhxmq3up2w0tamsplnna3wm0dkxnyu5x8k03sux5d63\""
  - - aa
    - "false"
  - - ab
    - "{\"name\":\"Default Abstract Account\"}"
  - - af
    - "[\"mock1ds5m6wwuu0cr35kmhxmq3up2w0tamsplnna3wm0dkxnyu5x8k03sux5d63\"]"
  - - ag
    - "{\"trace\":\"local\",\"seq\":1}"
  - - ah
    - "[[{\"info\":{\"namespace\":\"tester\",\"name\":\"mock-adapter1\",\"version\":{\"version\":\"1.0.0\"}},\"reference\":{\"adapter\":\"mock1ds5m6wwuu0cr35kmhxmq3up2w0tamsplnna3wm0dkxnyu5x8k03sux5d63\"}},null]]"
  - - ai
    - "[]"
  - - contract_info
    - "{\"contract\":\"abstract:account\",\"version\":\"0.24.1\"}"
  - - ownership
    - "{\"owner\":{\"monarchy\":{\"monarch\":\"mock1pgm8hyk0pvphmlvfjc8wsvk4daluz5tgrw6pu5mfpemk74uxnx9qwrtv4f\"}},\"pending_owner\":null,\"pending_expiry\":null}"
"abstract:ans-host":
  - - be
    - "[]"
  - - cfg
    - "{\"next_unique_pool_id\":1}"
  - - contract_info
    - "{\"contract\":\"abstract:ans-host\",\"version\":\"0.24.2\"}"
  - - ownership
    - "{\"owner\":\"mock14cl2dthqamgucg9sfvv4relp3aa83e40yfzzc8\",\"pending_owner\":null,\"pending_expiry\":null}"
"abstract:ibc-client":
  - - contract_info
    - "{\"contract\":\"abstract:ibc-client\",\"version\":\"0.24.1\"}"
  - - mod
    - "{\"module\":\"abstract:ibc-client\",\"version\":\"0.24.1\",\"dependencies\":[],\"metadata\":null}"
  - - ownership
    - "{\"owner\":\"mock14cl2dthqamgucg9sfvv4relp3aa83e40yfzzc8\",\"pending_owner\":null,\"pending_expiry\":null}"
"abstract:ibc-host":
  - - contract_info
    - "{\"contract\":\"abstract:ibc-host\",\"version\":\"0.24.1\"}"
  - - ownership
    - "{\"owner\":\"mock14cl2dthqamgucg9sfvv4relp3aa83e40yfzzc8\",\"pending_owner\":null,\"pending_expiry\":null}"
"abstract:module-factory":
  - - contract_info
    - "{\"contract\":\"abstract:module-factory\",\"version\":\"0.24.1\"}"
  - - ownership
    - "{\"owner\":\"mock14cl2dthqamgucg9sfvv4relp3aa83e40yfzzc8\",\"pending_owner\":null,\"pending_expiry\":null}"
"abstract:registry":
  - - "\u0000\u0002cb\u0000\u0006tester\u0000\tmock-app11.0.0"
    - "{\"app\":15}"
  - - "\u0000\u0002cb\u0000\u0006tester\u0000\tmock-app12.0.0"
    - "{\"app\":16}"
  - - "\u0000\u0002cb\u0000\u0006tester\u0000\rmock-adapter11.0.0"
    - "{\"adapter\":\"mock1ds5m6wwuu0cr35kmhxmq3up2w0tamsplnna3wm0dkxnyu5x8k03sux5d63\"}"
  - - "\u0000\u0002cb\u0000\u0006tester\u0000\rmock-adapter12.0.0"
    - "{\"adapter\":\"mock13nrd2qag9dww078z9a2way4kvzpr0k5dkv54e2xfmp2e2decfmjq4g06hw\"}"
  - - "\u0000\u0002cb\u0000\u0006tester\u0000\rmock-adapter21.0.0"
    - "{\"adapter\":\"mock1k9z7fvd3307ynapnk5gjffn706mz8qck502f7qwgt2a2a35kgqlqsv5ehm\"}"
  - - "\u0000\u0002cb\u0000\u0006tester\u0000\rmock-adapter22.0.0"
    - "{\"adapter\":\"mock1d2wr6ej95xepd3wmmpgrkyxwjns6gt5tfscrr3jcuetz7m7z0req77w6zq\"}"
  - - "\u0000\u0002cb\u0000\babstract\u0000\u0007account0.24.1"
    - "{\"account\":5}"
  - - "\u0000\u0002cb\u0000\babstract\u0000\bans-host0.24.2"
    - "{\"native\":\"mock1wx0qjtlz799pfxl73y2f4dv28nukztpmq2ztavk60v534tat9cds0pnytz\"}"
  - - "\u0000\u0002cb\u0000\babstract\u0000\bibc-host0.24.1"
    - "{\"native\":\"mock1aaf9r6s7nxhysuegqrxv0wpm27ypyv4886medd3mrkrw6t4yfcns7ctvz5\"}"
  - - "\u0000\u0002cb\u0000\babstract\u0000\bregistry0.24.3"
    - "{\"native\":\"mock1cag6cwygef6fuddgq2l44py7crpscufrza4mt3rum2axueemenmqapxshc\"}"
  - - "\u0000\u0002cb\u0000\babstract\u0000\nibc-client0.24.1"
    - "{\"native\":\"mock16jzpxp0e8550c9aht6q9svcux30vtyyyyxv5w2l2djjra46580wsgtnl4n\"}"
  - - "\u0000\u0002cb\u0000\babstract\u0000\u000emodule-factory0.24.1"
    - "{\"native\":\"mock1vnj802lyyupfaez73w7axfs3xveraxen370vhcnumaufj29r5rxqwxdk4n\"}"
  - - "\u0000\u0002ch\u0000\u0005local\u0000\u0000\u0000\u0000"
    - "\"mock1ys8clus4qf2haemyck0uljz74dj2k8hu4ggfpe7vty6zd5j2982q9sf90k\""
  - - "\u0000\u0002ch\u0000\u0005local\u0000\u0000\u0000\u0001"
    - "\"mock1jdl4tasw0x00k2edyf8656lqxy95gx06h2nt5n8qjzv45yxr5ekq0myk0q\""
  - - "\u0000\u0002cjabstract"
    - "{\"trace\":\"local\",\"seq\":0}"
  - - "\u0000\u0002cjtester"
    - "{\"trace\":\"local\",\"seq\":1}"
  - - "\u0000\u0002ck\u0000\u0005local\u0000\u0000\u0000\u0000"
    - "\"abstract\""
  - - "\u0000\u0002ck\u0000\u0005local\u0000\u0000\u0000\u0001"
    - "\"tester\""
  - - cfg
    - "{\"security_disabled\":true,\"namespace_registration_fee\":null}"
  - - ci
    - "2"
  - - contract_info
    - "{\"contract\":\"abstract:registry\",\"version\":\"0.24.3\"}"
  - - ownership
    - "{\"owner\":\"mock14cl2dthqamgucg9sfvv4relp3aa83e40yfzzc8\",\"pending_owner\":null,\"pending_expiry\":null}"
"tester:mock-adapter1":
  - - base_state
    - "{}"
  - - contract_info
    - "{\"contract\":\"tester:mock-adapter1\",\"version\":\"2.0.0\"}"
  - - mod
    - "{\"module\":\"tester:mock-adapter1\",\"version\":\"2.0.0\",\"dependencies\":[],\"metadata\":null}"
"tester:mock-adapter2":
  - - base_state
    - "{}"
  - - contract_info
    - "{\"contract\":\"tester:mock-adapter2\",\"version\":\"2.0.0\"}"
  - - mod
    - "{\"module\":\"tester:mock-adapter2\",\"version\":\"2.0.0\",\"dependencies\":[],\"metadata\":null}"
